### ISSUE TITLE
Tail do no consume input

### ISSUE THREAD

**Issue by albfan** [uutils/coreutils#1049]
On tail -f

Everything you type is not consumed and executed after stop command

**Comment by albfan** [comment]
Solved on #1048

**Comment by Minoru** [comment]
Can you provide a list of steps to reproduce this? Here's what I came up with, but maybe I'm just misunderstanding the issue.

1. In one terminal, run the following:

    ```shell
    $ echo hello | ( tail -f /etc/rc.local ; md5sum )
    ```

2. In another shell, kill the `tail`:

    ```shell
    # Find out the PID
    $ ps aux | grep 'tail -f /etc/resolv.conf' | grep -v grep
    minoru   13808  0.0  0.0 124800   460 pts/11   S+   15:37   0:00 tail -f /etc/resolv.conf
    $ kill 13808
    ```
3. In the first shell, observe the output. Mine is:

    ```
    # Generated by resolvconf
    nameserver 192.168.0.1
    options edns0

    b1946ac92492d2347c6235b4d2611184  -
    ```

    Note the last line. This is an MD5 hashsum of string "hello\n". This means `tail` didn't consume that "hello", and it went on to `md5sum`.

This is with `tail (GNU coreutils) 8.26`. Running the same experiment with previous `master` commit, ee9013215bed626fd14bfbcf73d3d90cc65afade, yields the same result, so uutils implementation is correct in this regard.

Am I missing or misunderstanding something?

**Comment by albfan** [comment]
I can reproduce tail consuming input this way:

    $ tail -f /etc/rc.local; cat -A

from another terminal:

    $ TAIL_PID=$(ps -ef | grep tai[l] | awk '{print $2}')
    $ echo -e 'rm\n\nmv\n\n\n\n' > /proc/$TAIL_PID/fd/0
    $ kill -2 $TAIL_PID

I guess those explains by itself, anyway it doesn't hurt little explanation:

- `$TAIL_PID` grabs pid for running tail
- `echo -e` send harming commands to stdin of running tail
- `kill -2` send a `SIGINT` (`Ctrl+C`) to tail
- `cat -A` echo output with special symbols for special characters like $ = \n

I can see input echoed while tail is running

```
rm

mv




```

but those lines are not passed to next command `cat`

I can dig into tail implementation to see if this consumption is really done or not anyway

What I'm looking for is a tail implementation that support this workflow:

- `tail -f` on a bunch of log files
- press `Enter` several times (to clean screen)
- write some text as note ("looking to find here blablabla that is failing when blablabla"
- stop tail with `Ctrl+C`
- do not see command line executing several `Enter` and my random notes as commands, specially if they become harming commands

**Comment by Minoru** [comment]
I ran your commands and indeed it all works as you describe. But running them with uutils ee9013215bed626fd14bfbcf73d3d90cc65afade works the same way, too!

It looks like comparing GNU and uutils implementations will be necessary after all.

**Comment by Minoru** [comment]
I replaced `tail` in your examples with `rev` (which reverses the strings in the input, e.g. turns "hello" into "olleh"), and sure enough, doing `echo "hello\n\n\nworld` resulted in the following output from `rev`:

```
hello


world
```

[This StackOverflow answer](https://stackoverflow.com/a/5380763/2350060) explains why this happens. Indeed, when I replaced `rev` with `cat | rev` and `echo`ed into `rev`'s FD 0, I got a reversed output:

```
olleh


dlrow
```

In other words, your example doesn't really reproduce the workflow you describe at the end; you should be doing `cat | ( tail -f /etc/rc.local; cat -A)` instead.

This doesn't explain why uutils and GNU tail behave differently for you, though. The only suggestion I have is to run `cargo clean && cargo build` to ensure you have a clean build, and ensuring you've build the latest commit (which is ee9013215bed626fd14bfbcf73d3d90cc65afade).

**Comment by albfan** [comment]
I should say this problem arises on windows, and other tail versions suffer that too. My patch fixes that, behaving like *nix systems. Maybe check if it is running under windows is an option to apply.

Will try your hints and look for causes, but is true that using it, rust tail will be ahead of the command it tries to mimic, so maybe it should not be merged until a that fix is applied on upstream projects

**Comment by Minoru** [comment]
Oh, Windows! This might depend on the shell as well; what are you running? (I've tested on Linux with ZSH and Bash.)

I don't have Windows, so I won't be able to help you there.

I'd also like to note that the tests in #1048 failed, even on Windows, and testing that PR locally, I can confirm that `tail -f` doesn't follow the file at all—it seems like it simply hangs until killed.

**Comment by albfan** [comment]
I use MINGW (from git bash for windows) Tested every possible tail implementation for windows and all of them keeps input for next command

Humm, seems it breaks *nix systems, so a check for OS running makes sense.

I will close this as it is clearly a new feature of windows OS. Any hint about how to implement `Ok(_)` so it uses datum but do not print anything? i read that result is annotated with `must_use` but don't know how to use it without doing nothing
